libs/api/protocol/src/lib/infra/service/unit-input.transformer.ts

```diff
@@ -0,0 +1,24 @@
+import { plainToInstance } from 'class-transformer';
+
+import {
+	RegisteredUnit,
+	UnknownUnit,
+} from '../../core/entity/partials/unit-partial.entity';
+import { UnitInput, UnitInputType } from '../view-model/unit-input.view-model';
+
+export class UnitInputTransformer {
+	static async transform(
+		input: UnitInput,
+	): Promise<RegisteredUnit | UnknownUnit> {
+		await input.validOrThrow();
```

Comment:
why validate in the transformer? is that a common task of the transformer ?

Reply:
As the `UnitInput` class basically is a superset of two classes, the transformer has to ensure the class is filled with data of either one.